### PR TITLE
Fix Narwhal Dockerfile in Sui repo

### PR DESCRIPTION
**Pull request by mwtian**
This needs to be tested after https://github.com/MystenLabs/sui-operations/pull/478 is merged.

### CODE REVIEW

**Review by akichidis — COMMENTED** [review]
@mwtian have you tried testing building the image locally? I saw the comment about the sui-operations but let's also make sure the image can be also built successfully with the docker-compose file/scripts we have.

Now when I try to build the image from the SUI root dir like this:
```
docker build -f narhwal/Docker/Dockerfile .
```

it complains about missing features `celo, benchmark` when building the binary (which are Narwhal features). Didn't have the opportunity to look deeper but was curious to see your results.

**Comment by mwtian** [comment]
> @mwtian have you tried testing building the image locally? I saw the comment about the sui-operations but let's also make sure the image can be also built successfully with the docker-compose file/scripts we have.

Fixed now.

File: narwhal/Docker/Dockerfile [hunk 1]
```diff
@@ -86,17 +86,17 @@ RUN cargo build --${BUILD_MODE} --features ${FEATURES} --bin node --bin benchmar
 FROM debian:bullseye-slim
 ARG BUILD_MODE
 
-WORKDIR "$WORKDIR/narwhal"
+WORKDIR "$WORKDIR/sui"
 
 # Copy the Narwhal node binary to bin folder
-COPY --from=builder narwhal/target/${BUILD_MODE}/node bin/
+COPY --from=builder sui/target/${BUILD_MODE}/node bin/
```
Comment (allan-bailey) [inline narwhal/Docker/Dockerfile:92]: We should rename the binary to something more descriptive from 'node'.  Maybe 'narwhal-node' ?

Comment (mwtian) [reply narwhal/Docker/Dockerfile]: Sounds good. Fixed all the affected locations that I can find.

**Comment by mwtian** [comment]
Narwhal docker workflow passed: https://github.com/MystenLabs/sui-operations/actions/runs/3202105451

**Comment by mwtian** [comment]
Gentle ping?

**Review by kchalkias — APPROVED** [review]
As it seems to work locally, approved

**Pull request merged**